src/arch/arm/machine/gic_v2.c

```diff
@@ -160,6 +162,7 @@ void setIRQTrigger(irq_t irq, bool_t trigger)
     } else {
         gic_dist->config[index] &= ~BIT(offset + 1);
     }
+    gic_dist_config_cache[index] = gic_dist->config[index];
```

Comment:
nit: doing it like this makes you do an unnecessary hardware access, you could just set `gic_dist_config_cache[index]` like `gic_dist->config[index]` is set and do the bit operation on it directly.